xmlcommand/BESXMLInterface.cc

```diff
@@ -315,8 +314,8 @@ void BESXMLInterface::log_the_command()
  */
 void BESXMLInterface::execute_data_request_plan()
 {
-    vector<BESXMLCommand *>::iterator i = d_xml_cmd_list.begin();
-    vector<BESXMLCommand *>::iterator e = d_xml_cmd_list.end();
+    auto i = d_xml_cmd_list.begin();
```

Comment:
This might be pedantic, but the loop could be rewritten as

for (const auto &cmd: d_xml_cmd_list) {
     cmd.prep_request();
      ...
}